framework/src/geomsearch/SlaveNeighborhoodThread.C

```diff
@@ -85,12 +83,14 @@ SlaveNeighborhoodThread::operator()(const NodeIdRange & range)
     for (unsigned int i = 0; i < patch_size; ++i)
       neighbor_nodes[i] = _trial_master_nodes[return_index[i]];
 
+    processor_id_type processor_id = _mesh.processor_id();
```

Comment:
I'd prefer if you grabbed this processor's ID directly from libMesh::ParallelObject (which just happens to be inherited by every object). Just call "processor_id()". Getting this processor off the mesh doesn't seem right even though it **should** be correct in all cases. Getting it from the current object is logically a better place.

Reply:
I tried calling processor_id() instead of _mesh.processor_id() from SlaveNeighborhoodThread.C and NearestNodeLocator.C and I am getting an error saying `use of declared identifier 'processor_id'`. I included libmesh/parallel.h and I am still getting the same error. Should I be including some other file for this to work?

Reply:
I still haven't figured out how to change `_mesh.processor_id()` to `processor_id()` without generating an error. I have addressed all the other comments.

Reply:
What error? Also, something in here is breaking Relap-7. We'll have to take a look at that.

Reply:
When I change _mesh.processor_id() to processor_id() in SlaveNeighborhoodThread.C, I get
```SlaveNeighborhoodThread.C:86:50: error: called object type 'processor_id_type'
      (aka 'unsigned short') is not a function or function pointer
    processor_id_type processor_id = processor_id();
                                     ~~~~~~~~~~~~^ ```

When I make the same change in NearestNodeLocator.C, I get
```NearestNodeLocator.C:201:53: error: use of undeclared identifier 'processor_id'
              _mesh.elemPtr(dof)->processor_id() != processor_id())
                                                    ^
```

I am getting the same error after including `libmesh/parallel.h`. So I am not sure if some other files need to be included before I can the processor_id() function. 

I do not have access to Relap-7. So, I am not sure what test is failing.

Reply:
You didn't get the actual error message there but I'm assuming something isn't defined. I'll see if I have a few mins where I can take a look at the Relap-7 error.